### PR TITLE
layers: Combine Discard Rectangle Mode/Enable VUs

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Based on https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/6953

### CODE REVIEW

File: tests/unit/shader_object.cpp [hunk 1]
```diff
@@ -1339,16 +1339,8 @@ TEST_F(NegativeShaderObject, InvalidExclusiveScissorCount) {
...
-    vk::EnumerateDeviceExtensionProperties(m_device->Physical(), nullptr, &count, nullptr);
-    std::vector<VkExtensionProperties> properties(count);
-    vk::EnumerateDeviceExtensionProperties(m_device->Physical(), nullptr, &count, properties.data());
-    for (const auto& p : properties) {
-        if (std::strcmp(p.extensionName, VK_NV_SCISSOR_EXCLUSIVE_EXTENSION_NAME) == 0) {
-            if (p.specVersion < 2) {
-                GTEST_SKIP() << "exclusiveScissor specVersion 2 not supported.";
-            }
-        }
+    if (!DeviceExtensionSupported(VK_EXT_DISCARD_RECTANGLES_EXTENSION_NAME, 2)) {
```
Comment (ziga-lunarg) [inline tests/unit/shader_object.cpp:1342]: `VK_NV_SCISSOR_EXCLUSIVE_EXTENSION_NAME` is being replaced with `VK_EXT_DISCARD_RECTANGLES_EXTENSION_NAME`, is this intended or copy paste error?

Comment (spencer-lunarg) [reply tests/unit/shader_object.cpp:1342]: 100% copy-and-paste error, will just fix this to be

```suggestion
    if (!DeviceExtensionSupported(VK_NV_SCISSOR_EXCLUSIVE_EXTENSION_NAME, 2)) {
```

(See I have one below as well)

Comment (spencer-lunarg) [reply tests/unit/shader_object.cpp]: fixed

**Review by ziga-lunarg — APPROVED** [review]
LGTM

**Pull request merged**